dlt/extract/utils.py

```diff
@@ -119,6 +119,29 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_generator(gen: Any) -> Any:
+    """Wraps an async generator into a list of awaitables"""
+    is_running = False
+    exhausted = False
+
+    async def run() -> TDataItems:
+        nonlocal is_running, exhausted
+        try:
+            return await gen.__anext__()
+        except StopAsyncIteration:
+            exhausted = True
+            raise
+        finally:
+            is_running = False
+
+    # it is best to use the round robin strategy here if multiple async generators are used in resources
+    while not exhausted:
+        while is_running:
+            yield None
```

Comment:
100% cpu usage here? this will be executed in tight loop right? if there's None we go for the next resource and we get None again so we go again etc.

btw. I think we are close and the idea with yielding None is good.